### ISSUE TITLE
Error / Internal Server Error

### ISSUE THREAD

**Issue by tobasium** [ciur/papermerge#581]
Hi i wantet to run papermerge but it did not work. Maybe someone could help me to solve the problem. 
Did not see where exactly the problem is.

I use portainer and tested 2 different[ compose files](https://docs.papermerge.io/3.0/setup/docker-compose/): 

Web App + Worker
PostgreSQL

Web App + Worker
for test i use the same data as in documentation.

papermerge starts

i try to login it loads and then it shows only error

i only see Error

Log in Docker:
sqlalchemy.exc.NoResultFound: No row was found when one was required

---------------------------------------------
PostgreSQL (change username tobasium)

papermerge starts

i try to login

Under the pw field it shows:
Internal Server Error

Docker Log:
[SQL: SELECT core_user.id AS core_user_id, core_user.username AS core_user_username, core_user.email AS core_user_email, core_user.password AS core_user_password, core_user.is_superuser AS core_user_is_superuser, core_user.is_staff AS core_user_is_staff, core_user.is_active AS core_user_is_active, core_user.first_name AS core_user_first_name, core_user.last_name AS core_user_last_name, core_user.home_folder_id AS core_user_home_folder_id, core_user.inbox_folder_id AS core_user_inbox_folder_id, core_user.date_joined AS core_user_date_joined, core_user.created_at AS core_user_created_at, core_user.updated_at AS core_user_updated_at 
FROM core_user 
WHERE core_user.username = %(username_1)s 
 LIMIT %(param_1)s]
[parameters: {'username_1': 'tobasium', 'param_1': 1}]
(Background on this error at: https://sqlalche.me/e/20/f405)

**Info:**
 - Papermerge Version [e.g. 3.0.1]

**Comment by ciur** [comment]
Can you please attach full webapp's docker log ?

Issue may be related (or event the same one) to [Issue#579](https://github.com/ciur/papermerge/issues/579). In case it is same problem - well, I am currently working on it!

**Comment by tobasium** [comment]
i will deploy both stacks and send you the log seperatly in 5 min

**Comment by tobasium** [comment]
only Web App + Worker

[_papermerge-web-1_logs.txt](https://github.com/ciur/papermerge/files/13997465/_papermerge-web-1_logs.txt)

**Comment by tobasium** [comment]
PostgreSQL compose

[_papermerge-web-1_logs (1).txt](https://github.com/ciur/papermerge/files/13997470/_papermerge-web-1_logs.1.txt)

**Comment by ciur** [comment]
In case of last logs there is an error:
```
sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) relation "core_user" does not exist
LINE 2: FROM core_user 
             ^
```
Which means that in your case tables were not created. Could you check if db `postgresql://scott:tiger@db:5432/mydatabase`  is accessible (i.e. you can connect to it) and it has some tables in it (e.g. `core_use`) ?

**Comment by tobasium** [comment]
could you quickly tell me how i could test it. 

here is the log from db
[_papermerge-db-1_logs.txt](https://github.com/ciur/papermerge/files/13997497/_papermerge-db-1_logs.txt)

**Comment by ciur** [comment]
I found the problem. The problem is that docker compose's "depends_on" waits only for the container to start, but not the service inside it; in other words "db" container may be up, but actual postgres/mysql database still may not be up.
I will fix that part. Thank you again for reporting the problem!

**Comment by ciur** [comment]
@tobasium 
I've published 3.0.2.
Also in documentation I've updated [docker-compose examples](https://docs.papermerge.io/3.0/setup/docker-compose/#mysql-mariadb) to include [healthcheck](https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck) key - which ensures that webapp/workers will start only after database process (not just db container) is up and running. Please use update docker compose examples.

**Comment by tobasium** [comment]
Hi, tested, and found new error:
[_papermerge-web-1_logs.txt](https://github.com/ciur/papermerge/files/14002088/_papermerge-web-1_logs.txt)

**Comment by tobasium** [comment]
Hi tested, Solr compose works. so that i can login into pm but solr container exits after start with following error:

Cannot write to /var/solr as 8983:8983
drwxr-xr-x 2 root root 2 Jan 21 17:02 /var/solr
Cannot write to /var/solr as 8983:8983
drwxr-xr-x 2 root root 2 Jan 21 17:02 /var/solr

**Comment by ciur** [comment]
I am closing the ticket as original issue was solved (original issue was that other app started before db service was up, which cause core_user table not found exception - this problem was solved in 3.0.2).
